### ISSUE TITLE
请问怎么保存成tensorflow serving 的pb文件？

### ISSUE THREAD

**Issue by efwfe** [gaussic/text-classification-cnn-rnn#115]
准备用tensorflow serving 进行部署，请问input_x 是需要在模型外进行处理才放到feed data 里面吗，另外，y_pred_cls这个变量怎么导入到pb图形中吗？
谢谢

**Comment by efwfe** [comment]
output 这个变量一直不知道怎么定义

**Comment by spartazhangzhang** [comment]
import os
import tensorflow as tf
from rnn_model import TextRNN, TRNNConfig

trained_checkpoint_prefix = './checkpoints/textrnn/best_validation'
export_dir = os.path.join('models',
                          '0')  # IMPORTANT: each model folder must be named '0', '1', ... Otherwise it will fail!
model = TextRNN(TRNNConfig())
loaded_graph = tf.Graph()
with tf.Session(graph=loaded_graph) as sess:
    # Restore from checkpoint
    loader = tf.train.import_meta_graph(trained_checkpoint_prefix + '.meta')
    loader.restore(sess, trained_checkpoint_prefix)

    input_x = tf.saved_model.utils.build_tensor_info(model.input_x)
    classification_outputs_classes = tf.saved_model.utils.build_tensor_info(model.prediction)
    classification_outputs_scores = tf.saved_model.utils.build_tensor_info(model.logits)
    keep_prob = tf.saved_model.utils.build_tensor_info(model.keep_prob)
    classification_signature = (
        tf.saved_model.signature_def_utils.build_signature_def(
            inputs={
                "input_x": input_x,
            },
            outputs= {
                "output_score": classification_outputs_scores,
                "predict_classification":
                    classification_outputs_classes,
            },

            method_name=tf.saved_model.signature_constants.CLASSIFY_METHOD_NAME))
    legacy_init_op = tf.group(tf.tables_initializer(), name='legacy_init_op')
    builder = tf.saved_model.builder.SavedModelBuilder(export_dir)
    builder.add_meta_graph_and_variables(sess,
                                         [tf.saved_model.tag_constants.TRAINING,
                                          tf.saved_model.tag_constants.SERVING],
                                         signature_def_map={
                                             tf.saved_model.signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY:
                                                 classification_signature},
                                         legacy_init_op = legacy_init_op
                                        )
    builder.save()
